groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/GroovyIndex.java

```diff
@@ -348,6 +358,25 @@ public Set<IndexedMethod> getMethods(final String name, final String clz, QueryS
         return methods;
     }
 
+    private boolean matchInsensitiveCamelCase(String text, String where) {
+        if (text.length() > where.length()) {
+            return false;
+        }
+        String textP = text.toLowerCase();
```

Comment:
Sorry :) I still believe this function is not a proper impl. At least two bugs here:
`matchInsensitiveCamelCase("RgExPars", "RegExParser")` produces `true`, but `RgEx` should not match `R`e`gEx`. The culprit is IMHO this lowercase, as we loose information on the Capital-and-lowercase exact substrings to be searched.

Try in IDE's Goto Type (type `RgExPars`)

The 2nd bug is 
```
if (whereP.length() <= (index + 1)) 
```
That causes same string match (`matchInsensitiveCamelCase("RegExParser", "RegExParser")`)  to return `false` -- the last matching character satisfies the condition.

Reply:
I have rewritten the algorithm. Added tests, so it should be ok now.